service/src/main/java/org/jboss/sbomer/service/feature/sbom/k8s/reconciler/GenerationRequestReconciler.java

```diff
@@ -420,44 +420,19 @@ public UpdateControl<GenerationRequest> reconcile(
         // 2 or more in case the generation is running
         Set<TaskRun> secondaryResources = context.getSecondaryResources(TaskRun.class);
 
-        UpdateControl<GenerationRequest> action = null;
-
         log.debug(
                 "Handling update for GenerationRequest '{}', current status: '{}'",
                 generationRequest.getMetadata().getName(),
                 generationRequest.getStatus());
 
-        switch (generationRequest.getStatus()) {
-            case NEW:
-                action = reconcileNew(generationRequest, secondaryResources);
-                break;
-            case INITIALIZING:
-                action = reconcileInitializing(generationRequest, secondaryResources);
-                break;
-            case INITIALIZED:
-                action = reconcileInitialized(generationRequest, secondaryResources);
-                break;
-            case GENERATING:
-                action = reconcileGenerating(generationRequest, secondaryResources);
-                break;
-            case FINISHED:
-                action = reconcileFinished(generationRequest, secondaryResources);
-                break;
-            case FAILED:
-                action = reconcileFailed(generationRequest, secondaryResources);
-                break;
-            default:
-                break;
-        }
-
-        // This would be unexpected.
-        if (action == null) {
-            log.error(
-                    "Unknown status received: '{}'' for GenerationRequest '{}",
-                    generationRequest.getStatus(),
-                    generationRequest.getMetadata().getName());
-            return UpdateControl.noUpdate();
-        }
+        UpdateControl<GenerationRequest> action = switch (generationRequest.getStatus()) {
+            case NEW -> reconcileNew(generationRequest, secondaryResources);
+            case INITIALIZING -> reconcileInitializing(generationRequest, secondaryResources);
+            case INITIALIZED -> reconcileInitialized(generationRequest, secondaryResources);
+            case GENERATING -> reconcileGenerating(generationRequest, secondaryResources);
+            case FINISHED -> reconcileFinished(generationRequest, secondaryResources);
+            case FAILED -> reconcileFailed(generationRequest, secondaryResources);
+        };
```

Comment:
I like this one! 👍🏻

Reply:
Nice one! We still need the 
```
        // This would be unexpected.
        if (action == null) {
            log.error(
                    "Unknown status received: '{}'' for GenerationRequest '{}",
                    generationRequest.getStatus(),
                    generationRequest.getMetadata().getName());
            return UpdateControl.noUpdate();
        }
```
part, correct?